testsuites/CBLTester/CBL_Functional_tests/cbl_log_decoder_tool/test_cbl_log_decoder.py

```diff
@@ -0,0 +1,183 @@
+            }
+        )
+        if status != 0:
+            raise Exception("Could not fetch cbl logs from windows ")
+
+
+def download_cbl_log(cbl_log_decoder_platform, liteserv_version, cbl_log_decoder_build):
+
+    version = liteserv_version.split('-')[0]
+    if cbl_log_decoder_platform == "windows":
+        package_name = "couchbase-lite-log-{}-{}-windows".format(version, cbl_log_decoder_build)
+    elif cbl_log_decoder_platform == "centos" or cbl_log_decoder_platform == "ubuntu" or cbl_log_decoder_platform == "rhel" or cbl_log_decoder_platform == "debian":
+        package_name = "couchbase-lite-log-{}-{}-centos".format(version, cbl_log_decoder_build)
+    else:
+        package_name = "couchbase-lite-log-{}-{}-macos".format(version, cbl_log_decoder_build)
+    package_zip_name = "{}.zip".format(package_name)
+    expected_binary_path = "{}/{}".format(BINARY_DIR, package_name)
+    if os.path.isfile(expected_binary_path):
+        log_info("Package is already downloaded. Skipping.")
+        return
+
+    # Package not downloaded, proceed to download from latest builds
+    downloaded_package_zip_name = "{}/{}".format(BINARY_DIR, package_zip_name)
+    if cbl_log_decoder_build is None:
+        try:
+            url = "{}/couchbase-lite-log/{}/{}".format(RELEASED_BUILDS, version, package_zip_name)
+        except Exception as err:
+            raise Exception(str(err) + "this version does not exist in release build, please provide cbl log decoder build number with this flag: --cbl-log-decoder-build")
+    else:
+        url = "{}/couchbase-lite-log/{}/{}/{}".format(LATEST_BUILDS, version, cbl_log_decoder_build, package_zip_name)
+
+    log_info("Downloading {} -> {}/{}".format(url, BINARY_DIR, package_zip_name))
+    resp = requests.get(url, verify=False)  # Need to resolve the certificate verification issue for release branch
+    resp.raise_for_status()
+    with open("{}/{}".format(BINARY_DIR, package_zip_name), "wb") as f:
+        f.write(resp.content)
+    extracted_directory_name = downloaded_package_zip_name.replace(".zip", "")
+    if cbl_log_decoder_platform == "macos":
+        with ZipFile("{}".format(downloaded_package_zip_name)) as zip_f:
+            zip_f.extractall("{}".format(extracted_directory_name))
+    else:
+        os.system("unzip {} -d {}".format(downloaded_package_zip_name, extracted_directory_name))
+    # Remove .zip
+    os.remove("{}".format(downloaded_package_zip_name))
+    os.chmod("{}/bin/cbl-log".format(extracted_directory_name), 0777)
+    return extracted_directory_name
+
+
+def decode_logs(extracted_directory_name, file):
+    output_file = "deps/binaries/cbl.txt"
+    command = "{}/bin/cbl-log logcat {} {}".format(extracted_directory_name, file, output_file)
+    return_val = os.system(command)
+    if return_val != 0:
+        raise Exception("{0} failed".format(command))
+    assert not is_binary(output_file), "{} decoded file is still in binary".format(file)
+
+
+def is_binary(filename):
```

Comment:
I googled it, seem this function will treat UTF-16 encoded text file as binary. You may add comments for it.
Once upgrading to python 3, this implementation need to be verified if it works.

Reply:
As it will be part of regression, once it is upgraded to python 3 , it will get tested.
I already added comments what this function do at the definition. Let me know if that is what you are asking for